### PR TITLE
initialized project structure

### PR DESCRIPTION
**Pull request by alekseypetrenko**

### CODE REVIEW

File: front/src/App.css [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+@keyframes App-logo-spin {
+  from {
+    transform: rotate(0deg);
+  }
+  to {
+    transform: rotate(360deg);
+  }
+}
```
Comment (andrew-kysel) [inline front/src/App.css]: думаю, это не нужно, если будем юзать ant design

File: front/src/App.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import React from "react";
+import "./App.css";
```
Comment (andrew-kysel) [inline front/src/App.js]: это тоже наверное лишнее

File: front/src/App.css (hunk 1 above)
Comment (alekseypetrenko) [reply front/src/App.css]: Deleted

File: front/src/App.js (hunk 2 above)
Comment (alekseypetrenko) [reply front/src/App.js]: done

File: back/app.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+const express = require("express");
+
+const app = express();
+
+// app.get("/", (req, res) => res.send("Hi"));
```
Comment (kolesnicknick) [inline back/app.js]: Change this file to typescript

File: back/config.json [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{
+    "db": {
+        "name": "lms",
+        "user": "postgres",
+        "password": "postgres",
+        "host": "127.0.0.1",
+        "port": 5432,
```
Comment (kolesnicknick) [inline back/config.json]: user, password, host, port should not be commited to the github repository but rather taken from environment variables, from 'dotenv' for example

File: back/db.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+	dialectOptions: {
+		supportBigNumbers: true
+	}
+});
+
+sequelize.authenticate()
+	.then(() => console.log("DB is connected"))
+	.catch(err => console.log(err));
+
+module.exports = sequelize;
```
Comment (kolesnicknick) [inline back/db.js]: file should end on empty line

File: front/package.json [hunk 6]
```diff
@@ -0,0 +1,38 @@
+{
+  "name": "front",
```
Comment (kolesnicknick) [inline front/package.json]: change project name to something that describes it better, lms-ui, for example

File: front/src/App.js [hunk 7]
```diff
@@ -0,0 +1,7 @@
+import React from "react";
+
+function App() {
+    return <div className="App">Learning managment system</div>;
```
Comment (kolesnicknick) [inline front/src/App.js]: typo in word management

File: package.json [hunk 8]
```diff
@@ -0,0 +1,36 @@
+{
+    "name": "dp186_lms",
```
Comment (kolesnicknick) [inline package.json]: this value should be discussed and updated I guess

File: package.json [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+    "version": "1.0.0",
+    "description": "Learning management system",
+    "main": "app.js",
```
Comment (kolesnicknick) [inline package.json]: this should be changed to the initial file, transpiled from typescript

**Review by kolesnicknick — CHANGES_REQUESTED** [review]

File: front/public/robots.txt [hunk 10]
```diff
@@ -0,0 +1,3 @@
+# https://www.robotstxt.org/robotstxt.html
+User-agent: *
+Disallow:
```
Comment (Giruta) [inline front/public/robots.txt]: мне кажется это дефолтный файлик не нужен

File: .gitignore [hunk 11]
```diff
@@ -0,0 +1,24 @@
+# See https://help.github.com/articles/ignoring-files/ for more about ignoring files.
+
+# dependencies
+/node_modules
+/front/node_modules
+/.pnp
+.pnp.js
+
```
Comment (antonpanchenk0) [inline .gitignore]: Mb add .idea folder for ide like WebStorm?

File: front/public/index.html [hunk 12]
```diff
@@ -0,0 +1,18 @@
+<!DOCTYPE html>
+<html lang="en">
+    <head>
+        <meta charset="utf-8" />
+        <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
+        <meta name="viewport" content="width=device-width, initial-scale=1" />
+        <meta name="theme-color" content="#000000" />
```
Comment (antonpanchenk0) [inline front/public/index.html]: Md delete this meta them-color?

File: back/app.js (hunk 3 above)
Comment (alekseypetrenko) [reply back/app.js]: done

File: back/config.json (hunk 4 above)
Comment (alekseypetrenko) [reply back/config.json]: done

File: back/db.js (hunk 5 above)
Comment (alekseypetrenko) [reply back/db.js]: done

File: front/package.json (hunk 6 above)
Comment (alekseypetrenko) [reply front/package.json]: done

File: front/src/App.js (hunk 7 above)
Comment (alekseypetrenko) [reply front/src/App.js]: done

File: package.json (hunk 9 above)
Comment (alekseypetrenko) [reply package.json]: done

File: front/public/robots.txt (hunk 10 above)
Comment (alekseypetrenko) [reply front/public/robots.txt]: done

File: .gitignore (hunk 11 above)
Comment (alekseypetrenko) [reply .gitignore]: done

File: front/public/index.html (hunk 12 above)
Comment (alekseypetrenko) [reply front/public/index.html]: done

File: .gitignore [hunk 13]
```diff
@@ -0,0 +1,24 @@
+# See https://help.github.com/articles/ignoring-files/ for more about ignoring files.
+
+# dependencies
+/node_modules
+/front/node_modules
```
Comment (staspitsyk) [inline .gitignore]: Are 4 and 5 lines redundant? Can just node_modules help to ignore all directories called node_modules in the current folder and any subfolders?
https://stackoverflow.com/questions/29820791/git-ignore-node-modules-folder-everywhere

Comment (alekseypetrenko) [reply .gitignore]: fixed

File: front/src/common/redux/actions/example.js [hunk 14]
```diff
@@ -0,0 +1,3 @@
+const applicationLoad = () => ({ type: "APP_LOAD" });
```
Comment (Alan18081) [inline front/src/common/redux/actions/example.js]: Use single quotes as much as possible

File: front/src/common/redux/reducers/index.js [hunk 15]
```diff
@@ -0,0 +1,29 @@
+const initialState = {
+    isLoad: false,
+    res: 0,
+};
+
+const reducer = (state = initialState, action) => {
+    switch (action.type) {
+        case "APP_LOAD":
```
Comment (Alan18081) [inline front/src/common/redux/reducers/index.js]: Please, move the action types to another file and export constants from there in the format
APP_LOAD = 'APP_LOAD'

File: front/src/components/exampleForRedux/ExampleForRedux.js [hunk 16]
```diff
@@ -0,0 +1,30 @@
...
+        const unsubscribe = store.subscribe(() => {
+            const check = store.getState();
+            if (check.isLoad) {
+                document.title = "React is loaded";
+                unsubscribe();
+            }
+        });
+    }
+    render() {
+        store.dispatch(actions.calculate("+", 180, 6));
```
Comment (Alan18081) [inline front/src/components/exampleForRedux/ExampleForRedux.js]: Please, don't import store directly, use `react-redux` for dispatching actions

Comment (alekseypetrenko) [reply front/src/components/exampleForRedux/ExampleForRedux.js]: done

File: front/src/common/redux/reducers/index.js (hunk 15 above)
Comment (alekseypetrenko) [reply front/src/common/redux/reducers/index.js]: done

File: front/src/common/redux/actions/example.js (hunk 14 above)
Comment (alekseypetrenko) [reply front/src/common/redux/actions/example.js]: will be fixed in scope of other task

**Review by Alan18081 — APPROVED** [review]

**Review by antonpanchenk0 — APPROVED** [review]

**Review by staspitsyk — APPROVED** [review]

**Review by OksanaKozhukh — APPROVED** [review]

**Pull request merged**